### PR TITLE
Fix Swift assertion for chunked upload

### PR DESCRIPTION
**Pull request by kleesc**
- Allow any binary stream (IOBase, including Werkzeug's LimitedStream) to Swift's type assertion
- Allow LimitingStream from util.registry.filelike to seek backward, since it is required by the Swift client in order to retry operations, if it is configured to do so

**Issue:** https://issues.redhat.com/browse/PROJQUAY-1508

**Changelog:** 
- Update use of `_pyio` to `io` (`io` is implemented in C instead of pure Python) in Swift's implementation
- Fix Swift storage allowed stream types
- Fix Swift storage connection retries

**Docs:** 

**Testing:** 

**Details:**

### CODE REVIEW

File: util/registry/filelike.py
Comment (alecmerdler) [inline util/registry/filelike.py:29]: Typo

Comment (kleesc) [reply util/registry/filelike.py]: Fixed

**Review by alecmerdler — DISMISSED** [review]

**Comment by kleesc** [comment]
@alecmerdler Can you re-approve? I rebased on master after merging #676 that was blocking CI build step.

**Review by alecmerdler — APPROVED** [review]

**Pull request merged**
